File: JAVA/GATE SIX/TestMiniParkingLot.java

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;


public class TestMiniParkingLot{

    MiniParkingLot park;
    
    @BeforeEach
    void setUp(){
    
        park = new MiniParkingLot();
    }    
        
    @Test
    public void testThatParkingSlotIsUp20(){
    
        int [][] lot = {{0,0,0,0,0,0,0,0,0,0},{0,0,0,0,0,0,0,0,0,0}};
        int expected = park.countParkingLot(lot); 
        int actual= 20;
        
        assertEquals(actual,expected);
    }
    
    
    @Test
    public void testThatAllLotsAtDayStartIsAvailable(){
    
        int [][] lot = {{0,0,0,0,0,0,0,0,0,0},{0,0,0,0,0,0,0,0,0,0}};
        boolean expected = park.checkAllLotIsEmptyAtDayStart(lot); 
        boolean actual= true;
        
        assertEquals(actual,expected);
    }
    
    @Test
    public void testThatCarAutoMaticallyParksAtLot1WhenLotNumberIsNotSpecified(){
    
        int [][] lot = {{0,0,0,0,0,0,0,0,0,0},{0,0,0,0,0,0,0,0,0,0}};
        final String command = "assign";
        int [][] expected = park.parkACarBySequence(lot,command); 
        int [][] actual= {{1,0,0,0,0,0,0,0,0,0},{0,0,0,0,0,0,0,0,0,0}};
        
        assertArrayEquals(actual,expected);
    }
    
    @Test
    public void testThatCarParksWithLotNumberSpecified(){
    
        int [][] lot = {{0,0,0,0,0,0,0,0,0,0},{0,0,0,0,0,0,0,0,0,0}};
        int lotNumber = 10;
        String command = "assign";
        int [][] expected = park.parkCarByLotNumber(lot, lotNumber, command); 
        int [][] actual= {{0,0,0,0,0,0,0,0,0,1},{0,0,0,0,0,0,0,0,0,0}};
        
        assertArrayEquals(actual,expected);
    }
    
    @Test
    public void testThatCarCanUnParksWithLotNumberSpecified(){
    
        int [][] lot = {{1,0,0,0,0,0,0,0,0,1},{0,0,0,0,0,0,0,0,0,0}};
        int lotNumber = 10;
        String command = "unassign";
        int [][] expected = park.unparkCarByLotNumber(lot,lotNumber,command); 
        int [][] actual= {{1,0,0,0,0,0,0,0,0,0},{0,0,0,0,0,0,0,0,0,0}};
        
        assertArrayEquals(actual,expected);
    }
    
    
}

File: PYTHON/GATE SIX/test_mini_parking_lot.py
from unittest import TestCase
from mini_parking_lot import *


class TestFile(TestCase):

    def test_that_parking_lot_has_20_slots(self):
    
        lot = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
        
        expected = 20
        actual = count_parking_lot(lot)
        
        self.assertEqual(actual,expected)
        
        
    def test_that_parking_lot_is_empty(self):
    
        lot = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
        
        expected = True;
        actual = check_all_lot_is_empty(lot)
        
        self.assertTrue(actual,expected)
        
    def test_that__lot_is_full(self):
    
        lot = [1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]
        
        expected = True;
        actual = check_that__lot_is_full(lot)
        
        self.assertTrue(actual,expected)
        
    def test_that_a_car_is_parked_by_sequence_when_lot_number_is_not_specified(self):
    
        lot = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
        command = "assign";
        expected = [1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0];
        actual = car_is_parked_by_sequence_when_lot_number_is_not_specified(lot, command)
        
        self.assertEqual(actual,expected)

